src/handles/pattern-handle.cpp

```diff
@@ -154,23 +89,69 @@ void PatternHandle::onInterest(const shared_ptr<const ndn::Name> &prefix,
         return;
     }
 
-    shared_ptr<IFetchPattern> p;
-    auto patIt = patterns_.find(patternName.get(0));
-    // If p is not found, 
+    auto patIt = patterns_.find(fetchPrefix);
+    // If p is found, 
+    // Result code: 402:=Duplicated fetch request
+    if(patIt != patterns_.end())
+    {
+        negativeReply(*interest, 402);
+        return;
+    }
+
+    // Else try to create specified pattern
+    shared_ptr<IFetchPattern> pattern = patternFactory_.create(
+        patternName, getFace(), getKeyChain(),
+        bind(static_cast<void(StorageEngine::*)(const ndn::Data&)>(&StorageEngine::put),
+             &getStorageHandle(), _1));
+
+    // If pattern can not be created, 
     // Result code: 404:=No such pattern is known
-    if(patIt == patterns_.end()){
+    if(pattern == nullptr)
+    {
         negativeReply(*interest, 404);
         return;
     }
-    p = patIt->second;
 
-    if (p)
+    pattern->fetch(fetchPrefix);
+    patterns_[fetchPrefix] = pattern;
+    this->onDataInsertion(fetchPrefix);
+
+    // reply with only status code, not negative though
+    negativeReply(*interest, 100);
```

Comment:
does it have to be renamed into something more general, like `sendReply` or similar (since it's used to send positive replies as well)?

Reply:
> does it have to be renamed into something more general, like `sendReply` or similar (since it's used to send positive replies as well)?

Maybe something like replyWithStatusCode works.